src/App.js

```diff
@@ -1,26 +1,82 @@
 import React from 'react';
 import logo from './logo.svg';
 import './App.css';
+import allContacts from './contacts.json'
+import Contacts from './components/Contact';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends React.Component {
+
+  state = {
+    firstContacts: allContacts.slice(0,5)
+  }
+
+  addRandomContact = () => {
+    const selectedRandomContact = allContacts[Math.floor(Math.random()*allContacts.length)];
+    this.setState({
+        firstContacts: [...this.state.firstContacts, selectedRandomContact]
+    });
+  }
+
+  sortContact = () => {
+    const sortedNameList = this.state.firstContacts.sort(function (a, b) {
+      if (a.name < b.name) {
+        return -1
+      } else if (a.name > b.name) {
+        return 1
+      } else {
+        return 0
+      }
+    })
```

Comment:
Aquí esto también te valdría y así vas probando cosas nuevas....

const sortedNameList = this.state.firstContacts.sort((a, b) => a.name.localeCompare(b.name))